### PR TITLE
now choosing an initial zoom value which fits the entire graph in the canvas

### PR DESCRIPTION
**Pull request by MaderNoob**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -86,6 +87,41 @@ function sortCoords() {
...
+    }))
+    // the desired zoom value to make the whole height of the graph fit inside the canvas.
+    const heightZoomValue = canvas.height / maxPointHeight
+
+    // we want to choose the smaller one of the 2 zoom values that we found so that both the width 
+    // and the height fit.
+    const zoomValue = Math.min(widthZoomValue, heightZoomValue)
+
+    coords = coords.map(([x, y]) => {
+        // to zoom the y coordinates we want to multiply the height of each point.
```
Comment (nickBes) [inline index.js:110]: why do we need to multiply the height if we don't do that in our zooming functionality. what difference does it make?

**Review by nickBes — CHANGES_REQUESTED** [review]
Zooming isn't enough because it only assures that the graph would fit the canvas if it'd be in the center of the screen, but it isn't. So we also need to move the graph to the center

File: index.js (hunk 1 above)
Comment (MaderNoob) [reply index.js:110]: Multiplying the height is like zooming out from the bottom left point. If i were to zoom out from the top-left, i would just multiply the y value, but i want to zoom out from the origin, which is bottom left, so i must mutliply the height instead.

File: index.js [hunk 2]
```diff
@@ -86,6 +90,93 @@ function sortCoords() {
...
+// makes sure that all of the given points are in the screen by zooming and moving the graph
+// the nonZoomablePoints are points that should also be included in the screen, but should not
+// be affected by the zoom.
+function includePointsInScreen(points) {
+    // found a boundary which contains all points
+    let boundaries = foundBoundariesOfPoints(points)
+
+    // find the dimensions of the boundary around all the points
+    const width = Math.abs(boundaries.right - boundaries.left);
+    const height = Math.abs(boundaries.bottom - boundaries.top);
```
Comment (nickBes) [inline index.js:129]: same here but switch the order to make it `top - bottom`

File: index.js [hunk 3]
```diff
@@ -86,6 +90,93 @@ function sortCoords() {
...
+
+    // the desired zoom value to make the whole height of the boundary fit inside the canvas.
+    const heightZoomValue = canvas.height / height
+
+    // we want to choose the smaller one of the 2 zoom values that we found so that both the width 
+    // and the height fit.
+    const zoomFactor = Math.min(widthZoomValue, heightZoomValue)
+
+    mapAllPoints((point) => zoomPoint(point, zoomFactor, origin))
+    points = points.map((point) => zoomPoint(point, zoomFactor, origin))
```
Comment (nickBes) [inline index.js:142]: why are you mapping the points twice? just don't pass the points array in the method.

File: index.js [hunk 4]
```diff
@@ -86,6 +90,93 @@ function sortCoords() {
...
+
+    // move the points so that all of them are in the screen
+    const boundariesAfterZoom = foundBoundariesOfPoints(points)
+    const movement = [-boundariesAfterZoom.left, -(boundariesAfterZoom.bottom - canvas.height)]
```
Comment (nickBes) [inline index.js:146]: maybe i'd explain why you make these calculations

File: index.js (hunk 2 above)
Comment (MaderNoob) [reply index.js:129]: no, bottom > top

**Pull request merged**